### ISSUE TITLE
note_names: Do not duplicate note name annotations on subsequent runs

### ISSUE THREAD

**Issue by westurner** [musescore/MuseScore#23502]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

Note names are duplicated when 'Plugins > Composing/Arranging tools > Note names' is run more than once

### Steps to reproduce

1. Enable the note_names plugin
2. 'Plugins > Composing/Arranging tools > Note names' (first)
3. 'Plugins > Composing/Arranging tools > Note names' (second)

### MuseScore Version

4.3.2

### Regression

I don't know

### Operating system

Fedora

### Additional context

Possible solutions:
- [ ] remove all text annotations that were added by note_names prior to running 
- [ ] to preserve manually-modified notations, add a ui checkbox and text to indicate that all existing note_names will be erased

**Comment by westurner** [comment]
Aren't these note names the notes necessary for chords:
- https://github.com/musescore/MuseScore/issues/23450

**Comment by Jojo-Schmitz** [comment]
> Aren't these note names the notes necessary for chords:
> 
> * [Allow to put chord diragrams at the top of a score  #23450](https://github.com/musescore/MuseScore/issues/23450)

No.

Other that that this is not a bug, but by design.
So you'd need a feature request rather than a bug report

**Comment by westurner** [comment]
In a different preparation, they could be.

E.g. BespokeSynth has a "show chord intervals" feature. Is that already
possible with musescore with a different plug-in?

Sometimes the interval between two notes is a Harmonic.

IMHO,
The "display chords at the top" task overlaps with the "identify intervals
between notes played simultaneously" task.

Perhaps the intervals would be useful to annotate in addition to the notes
and chords

**Comment by Jojo-Schmitz** [comment]
No, "allow chord diagrams above score" is very certainly unrelated

**Comment by MarcSabatella** [comment]
Fret diagrams are entirely unrelated to note heads.  And the usual way to add note heads to noteheads is not via a plugin, it's using the built-in feature on the Properties panel or in Staff/Part properties.

If you have further questions on how to use MsueScore Studio, please ask on the official support forum at musescore.org.  This issue tracker is meant for actual bugs, or specific feature requests (such as the consensus that might result from discussion on the forums).
